x/datadeal/types/deal.go

```diff
@@ -8,10 +8,12 @@ import (
 	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
 )
 
+const NonceSize = 12
+
 func NewDeal(dealID uint64, msg *MsgCreateDeal) *Deal {
 
 	dealAddress := NewDealAddress(dealID)
-
+	nonce := make([]byte, 12)
```

Comment:
I know that nonce will be generated when make a new deal. Or does the `create_deal` file include nonce?

Reply:
The nonce seems to be included because it calls `NewDeal` inside the `CreateDeal` keeper function.

Reply:
If the `nonce` is included in file, does the nonce need to change to `msg.Nonce`?

Reply:
I also agree with @inchori . It would be better to input nonce from Msg.

Reply:
I add `ReadFull` function to make nonce randomly. Thanks! 👍

Reply:
As @gyuguen said, I move that to msg. Please refer to commit [9a1ed93](https://github.com/medibloc/panacea-core/pull/433/commits/9a1ed9300a9379fdc2c2341850a51b5846229ee4).